cruise-control/src/main/java/com/linkedin/kafka/cruisecontrol/KafkaCruiseControl.java

```diff
@@ -163,6 +164,7 @@ private boolean isKafkaAssignerMode(List<String> goals) {
     ModelCompletenessRequirements modelCompletenessRequirements =
         modelCompletenessRequirements(goalsByPriority.values()).weaker(requirements);
     try (AutoCloseable ignored = _loadMonitor.acquireForModelGeneration(operationProgress)) {
+      sanityCheckBrokerPresence(brokerIds);
```

Comment:
Can we move `sanityCheckBrokerPresence(brokerIds);` out of the try-with-resources statement?

Reply:
Since we are calling `_loadMonitor.refreshClusterAndGeneration()` in this method, I think the lock is needed.